src/scraper.rs

```diff
@@ -226,6 +226,94 @@ impl Scraper {
         }
     }
 
+    fn handle_css(
+        scraper: &Scraper,
+        transmitter: &Sender<(Url, i32, i32)>,
+        url: &Url,
+        depth: i32,
+        ext_depth: i32,
+        data: &[u8],
+        http_charset: Option<String>,
+    ) -> Vec<u8> {
+        let charset_source_str = match Self::find_charset(data, http_charset) {
+            Some(s) => s,
+            None => {
+                warn!("Charset not found for {}, defaulting to UTF-8", url);
+                String::from("utf-8")
+            }
+        };
+
+        let need_charset_conversion = Self::needs_charset_conversion(&charset_source_str);
+
+        let charset_source = match encoding_rs::Encoding::for_label(charset_source_str.as_bytes()) {
+            Some(encoder) => encoder,
+            None => {
+                warn!(
+                    "Charset {} not supported for {}, defaulting to UTF-8",
+                    charset_source_str, url
+                );
+                encoding_rs::UTF_8
+            }
+        };
+        let charset_utf8 = encoding_rs::UTF_8;
+        let utf8_data = if need_charset_conversion {
+            Self::charset_convert(data, charset_source, charset_utf8)
+        } else {
+            Vec::from(data)
+        };
+        let source_path = match scraper.path_map.lock().unwrap().get(url.as_str()) {
+            Some(path) => path.clone(),
+            None => error!("Url {} was not found in the path map", url.as_str()),
+        };
+
+        let css_string = String::from_utf8_lossy(&utf8_data).to_string();
+        let url_regex = Regex::new(r"url\(([^)]+)\)").unwrap();
+
+        let urls: Vec<String> = url_regex
+            .captures_iter(&css_string)
+            .map(|url| url.get(1).unwrap().as_str().to_string())
+            .collect();
```

Comment:
Your are creating a vector from the captured elements. Later (L307) you call:
```rust
scraper.fix_domtree(&mut next_url, &source_path, &path);
```
where `&mut next_url` refers to an element of this vector of string. The problem is that you are not actually modifying the data, you are modifying a string in the vector of string.

Am I understanding this correctly? If yes, I think you can use the [replace method](https://docs.rs/regex/latest/regex/struct.Regex.html#method.replace) by implementing a [replacer](https://docs.rs/regex/latest/regex/trait.Replacer.html)

Reply:
You are right about that particular test. Check the following:
`cargo run https://www.purdue.edu/apps/account/cas/login?service=https%3A%2F%2Fwww.purdue.edu%2Fapps%2Fidphs%2FAuthn%2FExtCas%3Fconversation%3De1s1&entityId=https%3A%2F%2Ff81993d1-f040-40db-88cd-dddba8664daf.tenants.brightspace.com%2FsamlLogin --depth 2 --ext_depth 1 --delay 10 --random_range 5 --verbose` 
Note: some changes may need to be made to this command to avoid IP timeout.

You'll see that if you pull from this resource, since that particular css file respond with a 302 redirect, the path is not updated accordingly.